src/asynccontract.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { AsyncContract, Number } from '.';
+import { ValidationError } from './errors';
+
+describe('AsyncContract', () => {
+  describe('when function does not return a promise', () => {
+    it('throws a validation error', () => {
+      const contractedFunction = AsyncContract(Number).enforce(() => 7 as any);
+      expect(contractedFunction).toThrow(ValidationError);
+    });
+  });
+  describe('when a function does return a promise, but for the wrong type', () => {
+    it('throws a validation error asynchronously', async () => {
+      const contractedFunction = AsyncContract(Number).enforce(() => Promise.resolve('hi' as any));
+      await expect(contractedFunction()).rejects;
```

Comment:
I don't think this line tests what you want it to. I'm not sure if there's an easier way to do it with Jest's matchers, but this would get the job done:

```suggestion
      try {
        await contractedFunction();
        fail();
      } catch (e) {
        expect(e).toBeInstanceOf(ValidationError);
      }
```

Reply:
That definitely seems clearer regardless.